libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1801,6 +1804,16 @@ void AP_GPS::calc_blended_state(void)
     }
     timing[GPS_BLENDED_INSTANCE].last_fix_time_ms = (uint32_t)temp_time_1;
     timing[GPS_BLENDED_INSTANCE].last_message_time_ms = (uint32_t)temp_time_2;
+
+#ifndef HAL_BUILD_AP_PERIPH
+    if (timing[GPS_BLENDED_INSTANCE].last_message_time_ms > last_blended_message_time_ms) {
```

Comment:
Just use this as the first line of the if on line 1812, and remove the `data_should_be_logged` bool. It's just increasing the complexity here (and maybe the code size, but hopefully the compiler is managing this)

Reply:
Yep.